scripts/files-touched-check.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+import os
+import sys
+import subprocess
+
+travis_commit_range = os.getenv('TRAVIS_COMMIT_RANGE')
+if not travis_commit_range:
+    print("Travis commit range is empty, exiting...")
+    sys.exit(0)
+
+try:
+    result = subprocess.check_output(['git', 'diff', '--no-commit-id', '--name-status', '-r', travis_commit_range])
+except Exception as e:
+    print(e.output)
+    raise e
+
+files_added = result.decode('utf-8').splitlines()
+print(files_added)
+subdir_name = ""
+    
+for file_added in files_added:
+    file_added = file_added.split()
+
+    # Exclude certain files from some checks
+    if file_added[1].startswith("scripts/") or file_added[1] in ['README.md', '.travis.yml', '.gitattributes']:
+        print("Warning: modified non-gitian file", file_added[1])
+        continue
+
+    # Check that files are only added, not modified or deleted
+    if file_added[0] != 'A':
+        print("Error: modified or removed existing file:", file_added[1])
+        sys.exit(1)
+
+    # Check that files added are only added to a single subdirectory name
+    if file_added[1].count('/') > 1:
```

Comment:
As a bonus, this could also check:
- that the format of the top subdir matches a list of patterns (`<version>-linux` `<version>-osx-signed` `<version>-osx-unsigned` `<version>-win-signed` `<version>-win-unsigned` where version is simply a wildcard you could check with `.endswith()` or a re)
- that the depth (`len(file_added.split('/'))`) is 3

(not necessary in this PR, to be clear)